### PR TITLE
New fp8_transpose_dbias kernel

### PR DESCRIPTION
**Pull request by vasunvidia**
Signed-off-by: Vasudevan Rengasamy <vrengasamy@nvidia.com>

### CODE REVIEW

File: transformer_engine/common/transpose/transpose_fusion.cu [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+                                                             const int dbias_shfl_src_lane,
+                                                             const bool valid_store) {
+  using T = typename OVec::type;
+  using OVecC = Vec<T, nvec_in>;
+
+  CVec step_dbias; step_dbias.clear();
+
+#pragma unroll
+  for (unsigned int i = 0; i < nvec_out; ++i) {
+    OVecC out_cast;
```
Comment (timmoon10) [inline transformer_engine/common/transpose/transpose_fusion.cu:33]: Unused variable:
```suggestion
```

File: transformer_engine/common/transpose/transpose_fusion.cu [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+
+template <bool full_tile, int nvec_in, int nvec_out,
+          typename IVec, typename OVec, typename CVec, typename CType>
+inline __device__ void transpose_regs_partial_dbias(const IVec (&in)[nvec_out],
+                                                             OVec (&out_trans)[nvec_in],
+                                                             CVec &out_dbias,  // NOLINT(*)
+                                                             const size_t stride,
+                                                             const CType scale_inv,
+                                                             const int dbias_shfl_src_lane,
+                                                             const bool valid_store) {
```
Comment (timmoon10) [inline transformer_engine/common/transpose/transpose_fusion.cu:25]: Unused args:
```suggestion
template <int nvec_in, int nvec_out,
          typename IVec, typename OVec, typename CVec, typename CType>
inline __device__ void transpose_regs_partial_dbias(const IVec (&in)[nvec_out],
                                                             OVec (&out_trans)[nvec_in],
                                                             CVec &out_dbias,  // NOLINT(*)
                                                             const CType scale_inv,
                                                             const int dbias_shfl_src_lane) {
```

File: transformer_engine/common/transpose/transpose_fusion.cu [hunk 3]
```diff
@@ -0,0 +1,536 @@
...
+
+
+}  // namespace transformer_engine
+
+void nvte_fp8_transpose_dbias(const NVTETensor input,
+                               NVTETensor transposed_output,
+                               NVTETensor dbias,
+                               NVTETensor workspace,
+                               cudaStream_t stream) {
+  using namespace transformer_engine;
```
Comment (timmoon10) [inline transformer_engine/common/transpose/transpose_fusion.cu:530]: Missing NVTX marker:
```suggestion
  NVTE_API_CALL(nvte_transpose_dbias);
  using namespace transformer_engine;
```

**Review by timmoon10 — CHANGES_REQUESTED** [review]
Overall looks good, code is mostly the same as https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/common/transpose/cast_transpose_fusion.cu. The NVTX macro is most important suggestion. Removing some unused variables would also help clean up the code.

**Review by timmoon10 — APPROVED** [review]
LGTM

**Review by ksivaman — APPROVED** [review]

**Pull request merged**